tests/zkevm/test_worst_compute.py

```diff
@@ -955,3 +956,40 @@ def test_empty_block(
         post={},
         blocks=[Block(txs=[])],
     )
+
+
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.parametrize(
+    "empty_calldata",
+    [False, True],
+)
+def test_worst_calldataload(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    fork: Fork,
+    empty_calldata: bool,
+):
+    """Test running a block with as many CALLDATALOAD as possible."""
+    env = Environment()
+
+    code_prefix = Op.JUMPDEST
+    code_suffix = Op.PUSH0 + Op.JUMP
+    code_body_len = MAX_CODE_SIZE - len(code_prefix) - len(code_suffix)
+    code_loop_iter = Op.POP(Op.PUSH0 + Op.CALLDATALOAD)
```

Comment:
For more CALLDATALOAD, use empty calldata or calldata filled with zeros (arbitray length), then CALLDATALOAD will push 0 to stack in all cases and PUSH0/POP combo is not necessary :smile: :+1:

Reply:
So:

`PUSH0 JUMPDEST CALLDATALOAD CALLDATALOAD .. .. .. PUSH1 0x01 JUMP` would fill with more `CALLDATALOAD` iff calldata is consists only of zero bytes :)

Reply:
So:

`PUSH0 JUMPDEST CALLDATALOAD CALLDATALOAD .. .. .. CALLDATALOAD CALLDATALOAD PUSH1 0x01 JUMP` would fill with more `CALLDATALOAD` iff calldata is consists only of zero bytes :)

Reply:
Actually, just realized, if you want calldata you could even make it "loop" over it by using above code (without using explicit `PUSH0 CALLDATALOAD POP`) and use (for instance):
```
0x0000..0020
0x0000..0040
0x0000..0000
```

To make it `CALLDATALOAD(32) CALLDATALOAD(64) CALLDATALOAD(0)`

Reply:
And for loop balance to avoid stack overflows insert a POP before the JUMP

`PUSH0 JUMPDEST CALLDATALOAD CALLDATALOAD .. .. .. CALLDATALOAD CALLDATALOAD POP PUSH1 0x01 JUMP`

This would work (?) in all cases I think (based on max code size and your calldata input). Would avoid theoretical stack overflows

Reply:
That would use some extra gas from tx calldata, no? 
Is that POP before the `PUSH1`, needed?

I switched now to you first suggestion:
```
tests/zkevm/test_worst_compute.py::test_worst_calldataload[fork_Cancun-blockchain_test_from_state_test-empty_calldata_True]-1    2015521819
tests/zkevm/test_worst_compute.py::test_worst_calldataload[fork_Cancun-blockchain_test_from_state_test-empty_calldata_False]-1   2423216794
```

Reply:
Yes you are right! This would indeed work. I got a bit confused in my head with either stack imbalance (when jumping back to the JUMPDEST adding extra items) or having an invalid JUMP (based on the CALLDATALOAD jump on that). But the pattern you just provided should indeed work for all `calldata` patterns :smile: :+1: